packages/core/src/types/asyncapi.ts

```diff
@@ -0,0 +1,1141 @@
+    authorizationUrl: { type: 'string' },
+  },
+  required: ['authorizationUrl', 'scopes'],
+};
+
+const PasswordFlow: NodeType = {
+  properties: {
+    refreshUrl: { type: 'string' },
+    scopes: { type: 'object', additionalProperties: { type: 'string' } }, // TODO: validate scopes
+    tokenUrl: { type: 'string' },
+  },
+  required: ['tokenUrl', 'scopes'],
+};
+
+const ClientCredentials: NodeType = {
+  properties: {
+    refreshUrl: { type: 'string' },
+    scopes: { type: 'object', additionalProperties: { type: 'string' } }, // TODO: validate scopes
+    tokenUrl: { type: 'string' },
+  },
+  required: ['tokenUrl', 'scopes'],
+};
+
+const AuthorizationCode: NodeType = {
+  properties: {
+    refreshUrl: { type: 'string' },
+    authorizationUrl: { type: 'string' },
+    scopes: { type: 'object', additionalProperties: { type: 'string' } }, // TODO: validate scopes
+    tokenUrl: { type: 'string' },
+  },
+  required: ['authorizationUrl', 'tokenUrl', 'scopes'],
+};
+
+const SecuritySchemeFlows: NodeType = {
+  properties: {
+    implicit: 'ImplicitFlow',
+    password: 'PasswordFlow',
+    clientCredentials: 'ClientCredentials',
+    authorizationCode: 'AuthorizationCode',
+  },
+};
+
+const SecurityScheme: NodeType = {
+  properties: {
+    type: {
+      enum: [
+        'userPassword',
+        'apiKey',
+        'X509',
+        'symmetricEncryption',
+        'asymmetricEncryption',
+        'httpApiKey',
+        'http',
+        'oauth2',
+        'openIdConnect',
+        'plain',
+        'scramSha256',
+        'scramSha512',
+        'gssapi',
+      ],
+    },
+    description: { type: 'string' },
+    name: { type: 'string' },
+    in: { type: 'string', enum: ['query', 'header', 'cookie', 'user', 'password'] },
+    scheme: { type: 'string' },
+    bearerFormat: { type: 'string' },
+    flows: 'SecuritySchemeFlows',
+    openIdConnectUrl: { type: 'string' },
+  },
+  required(value) {
+    switch (value?.type) {
+      case 'apiKey':
+        return ['type', 'in'];
+      case 'httpApiKey':
+        return ['type', 'name', 'in'];
+      case 'http':
+        return ['type', 'scheme'];
+      case 'oauth2':
+        return ['type', 'flows'];
+      case 'openIdConnect':
+        return ['type', 'openIdConnectUrl'];
+      default:
+        return ['type'];
+    }
+  },
+  allowed(value) {
+    switch (value?.type) {
+      case 'apiKey':
+        return ['type', 'in', 'description'];
+      case 'httpApiKey':
+        return ['type', 'name', 'in', 'description'];
+      case 'http':
+        return ['type', 'scheme', 'bearerFormat', 'description'];
+      case 'oauth2':
+        return ['type', 'flows', 'description'];
+      case 'openIdConnect':
+        return ['type', 'openIdConnectUrl', 'description'];
+      default:
+        return ['type', 'description'];
+    }
+  },
+  extensionsPrefix: 'x-',
+};
+
+// --- Per-protocol node types
```

Comment:
Let's move those to a different files (maybe inside `async/` folder).

Reply:
I really wanted to put these in their own files, can you maybe move one to show how to do it and bring it into this structure correctly? @Roman didn't have time to help on this and I wasn't sure ...

Reply:
Let's keep it in this file for now. It can be always moved.